assignments/lambda-classes.js

```diff
@@ -1 +1,160 @@
 // CODE here for your Lambda Classes
+
+class Person {
+  constructor(props) {
+    this.name = props.name;
+    this.age = props.age;
+    this.location = props.location;
+    this.gender = props.gender;
+  }
+
+  speak() {
+    return `Hello my name is ${this.name}, I'm from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(props) {
+    super(props);
+    this.specialty = props.specialty;
+    this.favLanguage = props.favLanguage;
+    this.catchPhrase = props.catchPhrase;
+  }
+
+  demo(subject) {
+    return `Today we're learning ${subject}`;
+  }
+
+  grade(student, subject) {
+    return `${student.name} gets a perfect score on ${subject}`;
+  }
+}
+
+class ProjectManager extends Instructor {
+  constructor(props) {
+    super(props);
+    this.gradClassName = props.gradClassName;
+    this.favInstructor = props.favInstructor;
+  }
+
+  standUp(slackChannel) {
+    return `${this.name} announces to #${slackChannel}, @channel standy time!`;
+  }
+
+  debugsCode(student, subject) {
+    return `${this.name} debugs ${student.name}'s code on ${subject}`;
+  }
+}
+
+class Student extends Person {
+  constructor(props) {
+    super(props);
+    this.prevBackground = props.prevBackground;
+    this.className = props.className;
+    this.favSubjects = props.favSubjects;
+  }
+
+  listsSubjects() { // instructions for this method were not clear
+   // return this.favSubjects.join(', ');
```

Comment:
this one was a little vague. I think either one of these are acceptable. Nice use of forEach